<!-- -->
<launch>
    <!-- Camera that is going to be used -->
    <arg name="camera"/>

    <arg name="tf_x"/>
    <arg name="tf_y"/>
    <arg name="tf_z"/>
    <arg name="tf_qx"/>
    <arg name="tf_qy"/>
    <arg name="tf_qz"/>
    <arg name="tf_qw"/>

    <!-- base_link to camera tf -->
    <node ns="$(arg camera)" pkg="tf" type="static_transform_publisher" name="base_link_to_camera"
          args="$(arg tf_x) $(arg tf_y) $(arg tf_z) $(arg tf_qx) $(arg tf_qy) $(arg tf_qz) $(arg tf_qw) base_link $(arg camera) 10"/>

    <!-- camera_info -->
    <include ns="$(arg camera)" file="$(find adeye)/launch/camera_info_publisher.launch">
        <arg name="node_name" value="camera_info_publisher"/>
        <arg name="args_list" value="camera_info $(arg camera) 720 960 0.0 0.0 0.0 0.0 0.0 1125 0.0 480 0.0 1125 360 0.0 0.0 1.0 1.0 0.0 0.0 0.0 1.0 0.0 0.0 0.0 1.0 1125 0.0 480 0.0 0.0 1125 360 0.0 0.0 0.0 1.0 0.0"/>
    </include>

    <!-- vision_ssd_detect -->
    <include file="$(find adeye)/launch/vision_ssd_detect.launch">
        <arg name="camera" value="$(arg camera)"/>
        <arg name="use_gpu" value="true"/>
        <arg name="gpu_device_id" value="0"/>
        <arg name="score_threshold" value="0.5"/>
        <arg name="network_definition_file"
             value="$(env HOME)/AD-EYE_Core/AD-EYE/Data/ssdcaffe_models/AD-EYE_SSD_Model/SSD_512x512/deploy.prototxt"/>
        <arg name="pretrained_model_file"
             value="$(env HOME)/AD-EYE_Core/AD-EYE/Data/ssdcaffe_models/AD-EYE_SSD_Model/SSD_512x512/VGG_Dataset_SSD_512x512_iter_60000.caffemodel"/>
        <arg name="image_src" value="image_raw"/>
        <arg name="image_out" value="image_rects"/>
        <arg name="object_src" value="image_detector/objects"/>
    </include>

    <!-- camera_raw -->
    <include file="$(find adeye)/launch/new_image_raw.launch">
        <arg name="camera" value="$(arg camera)"/>
        <arg name="image_src" value="sim_image_raw"/>
        <arg name="image_out" value="image_raw"/>
    </include>

    <!-- range_vision_fusion -->
    <include file="$(find adeye)/launch/range_vision_fusion.launch">
        <arg name="camera" value="$(arg camera)"/>
        <arg name="detected_objects_range" value="detection/lidar_radar_fusion/objects"/>
        <arg name="detected_objects_vision" value="image_tracker/objects"/>
        <arg name="detected_objects_fusion" value="fusion/objects"/>
        <arg name="camera_info_src" value="/camera_info"/>
        <arg name="min_car_dimensions" value="[3,2,2]"/>
        <arg name="min_person_dimensions" value="[1,2,1]"/>
        <arg name="min_truck_dimensions" value="[4,2,2]"/>
        <arg name="sync_topics" value="false"/>
        <arg name="overlap_threshold" value="0.4"/>
    </include>

    <!-- vision_beyond_track-->
    <node ns="$(arg camera)" pkg="vision_beyond_track" name="vision_beyond_track" type="vision_beyond_track" output="screen">
        <param name="camera_info_src" type="str" value="/$(arg camera)/camera_info"/>
        <param name="objects_topic_src" type="str" value="/$(arg camera)/image_detector/objects"/>
        <param name="camera_height" type="double" value="$(arg tf_z)"/>
        <remap from="/detection/image_tracker/objects" to="/$(arg camera)/image_tracker/objects"/>
    </node>

</launch>
